implement create new task use case with test

hazemka opened the pull request:
- add new custom exception for empty values
- edit constructor parameter in TaskRepositoryImpl.kt to correct types
- fix error at LogRepositoryImpl.kt

hazemka committed:
implement create new task use case with test
- add new custom exception for empty values
- edit constructor parameter in TaskRepositoryImpl.kt to correct types
- fix error at LogRepositoryImpl.kt

AbdoTalaat74 commented on src/main/kotlin/logic/useCases/tasks/CreateNewTaskUseCase.kt:
```diff
@@ -1,4 +1,22 @@
+        if (validateTaskValues(task)){
+            taskRepository.createNewTask(task)
+        }else
+            throw EmptyValuesException("Task has empty values")
```
make the string as a constant in companion object

AbdoTalaat74 commented on src/test/kotlin/logic/useCases/tasks/CreateNewTaskUseCaseTest.kt:
```diff
@@ -0,0 +1,103 @@
+        // When & Then
+        assertThrows(EmptyValuesException::class.java){
+            createNewTaskUseCase.createNewTask(task)
+        }
```
use kotest for assertion

AbdoTalaat74 commented on src/test/kotlin/logic/useCases/tasks/CreateNewTaskUseCaseTest.kt:
```diff
@@ -0,0 +1,103 @@
+        // When & Then
+        assertThrows(EmptyValuesException::class.java){
+            createNewTaskUseCase.createNewTask(task)
+        }
```
use kotest here too

AbdoTalaat74 commented on src/test/kotlin/logic/useCases/tasks/CreateNewTaskUseCaseTest.kt:
```diff
@@ -0,0 +1,103 @@
+        // When & Then
+        assertThrows(EmptyValuesException::class.java){
+            createNewTaskUseCase.createNewTask(task)
+        }
```
kotest here too

hazemka committed:
change some lines in create new task
- change string for message exception to companion object
- use kotest for test

hazemka committed:
replace framework test to kotest
